### ISSUE TITLE
Linkerd stucks when namerd is hotswapped

### ISSUE THREAD

**Issue by leozc** [linkerd/linkerd#807]
Setup
1. Linkerd, namerd, consul1 and consul2 are hosted in different boxes.
2. namerd points to consul1 
3. consul1 has one datacenter called dc and consul2 has one datacenter called us-east-1a

Here are the steps to repo
1. update namerd to point to consul2, as well as the config accordingly
2. restart namerd.
3. From the namerd log, we still observer errors similar to below after even 1+ hours
> namerd     | E 1103 23:27:16.285 THREAD17: Retrying Consul request 'GET /v1/catalog/services?dc=dc1' on NonFatal error: UnexpectedResponse(500: No path to datacenter)
4. linkerd failed to resolve some URLs, I assume it is because the dc issue. 

Expectation:
after namerd restarting, I assume eventually (and briefly) linkerd get the latest config from namerd.

Workaround: 
Restart all linkerd -> problem solved.

**Comment by adleong** [comment]
I think this is due to namerd stamp collision when namerd is restarted.

**Comment by adleong** [comment]
I think we'll fix this by moving to a streaming API (either HTTP/1.1 or H2, possibly gRPC).

**Comment by Ashald** [comment]
1) IIRC there was a fix for Mux in `0.8.2` (we didn't see this issue since we updated)
2) To best of my knowledge as per now there is an HTTP/1.1 streaming API that can be used instead of Mux (which is state-less and shouldn't have such issues by design).

TL;DR: I think we can close this issue. :)

**Comment by leozc** [comment]
YES close :)

**Comment by Ashald** [comment]
From our experience, we stopped seeing such errors for real only after switching to `gRPC`.
